pkgs/applications/misc/xmrig/xmrig-cuda.nix

```diff
@@ -0,0 +1,46 @@
+{ stdenv
+, autoAddOpenGLRunpathHook
+, fetchFromGitHub
+, cmake
+, pkgs
+, tree
+, cudaArch ? 61
+}:
+
+let
+  libcudaDir = "/run/opengl-driver/lib";
+in
+stdenv.mkDerivation rec {
+  inherit (pkgs.cudaPackages) cudatoolkit cuda_cudart;
+
+  version = "6.17.0";
+  pname = "xmrig-cuda";
+  src = fetchFromGitHub {
+    owner = "xmrig";
+    repo = "xmrig-cuda";
+    rev = "v${version}";
+    sha256 = "sha256-vRVLmMgCG42btm0jcteZiftchjRZotBgT98znQV0I+k=";
+  };
+
+  nativeBuildInputs = [ cmake autoAddOpenGLRunpathHook ];
+
+  buildInputs = [
+    cudatoolkit
+    cuda_cudart
```

Comment:
Move `cuda_nvrtc` to over here. Also, upstream seems to be using NVidia's CUB library. That comes from `cuda_cccl`